### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by maov19**
### Changes Made:

- [ ] Created the Solver class to encapsulate the required functionality.

- [ ] Implemented the factorial method that calculates the factorial of a given positive integer N. It handles the special case where the factorial of 0 is 1 and raises an exception for negative integers.

- [ ] Added the reverse method to reverse a given string word.

- [ ] Implemented the fizzbuzz method, which returns a string based on the provided integer N following the FizzBuzz game rules.

### Testing:

- [ ] Created comprehensive tests for each method to cover different scenarios and edge cases.

- [ ] Ensured that all tests pass successfully and maintain good code coverage.

### CODE REVIEW

File: tests/tdd-tutorial/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  describe '#factorial' do
+    it 'returns 120' do
+      five = Solver.new
+      expect(five.factorial(5)).to eq(120)
+    end
+
+    it 'returns "No factorial for negative integers"' do
+      negative = Solver.new
+      expect(negative.factorial(-1)).to eq('No factorial for negative integers')
+    end
```
Comment (Owusu-Desmond) [inline tests/tdd-tutorial/solver_spec.rb]: - [ ] Please from the project requirement, if a negative integer is given it should raise an exception so your need to raise an exception not a to equal to a string :pray: 
like :point_down: 
`expect { resolver.factorial(-5) }.to raise_exception`

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @maov19 and @MICHAELKITH , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: tests/tdd-tutorial/solver_spec.rb (hunk 1 above)
Comment (maov19) [reply tests/tdd-tutorial/solver_spec.rb]: Fixed!

**Review by Tufahel — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Hi @maov19 & @MICHAELKITH :wave: 
 
Your project is complete now!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 👏🏻



:heavy_check_mark: Descriptive pull request.
:heavy_check_mark: Correct Git flow.
:heavy_check_mark: Implement all the requirements.

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
